### DISCUSSION TITLE
np.dot is faster than jnp.dot in docs

### DISCUSSION THREAD

**Discussion by rozeappletree** [jax-ml/jax#8673] [Q&A] [answered]
According to docs (https://jax.readthedocs.io/en/latest/notebooks/quickstart.html)

```
x = np.random.normal(size=(size, size)).astype(np.float32)
%timeit np.dot(x, x.T)
```
Takes
```
223 ms ± 804 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

which is faster than

```
from jax import device_put
x = np.random.normal(size=(size, size)).astype(np.float32)
x = device_put(x)
%timeit jnp.dot(x, x.T).block_until_ready()
```
```
401 ms ± 928 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Am I missing something ???

**Comment by jakevdp** [comment, marked as answer]
Thanks for the question. Agreed this is a bit confusing, but just before the timing you quote the docs say

> If you have a GPU (or TPU!) these calls run on the accelerator and have the potential to be much faster than on CPU.

The readthedocs documentation build uses a CPU, where, for simple microbenchmarks consisting of single array operations, numpy is generally faster. Perhaps we could add some language to the doc to clarify this.

**Reply by jakevdp** [reply]
I've added a section to the FAQ addressing this, because this kind of question comes up often: https://github.com/google/jax/pull/8681
